pkg/sources/sources.go

```diff
@@ -28,6 +29,71 @@ type Chunk struct {
 	Verify bool
 }
 
+// Cache used by Sources to help reduce duplicate data scans.
+type Cache interface {
+	Set(key, value string) error
+	Get(key string) (string, bool)
+	Delete(key string) error
+	Clear() error
+}
+
+// SourceConfigOption is responsible for configuring a SourceConfig.
+type SourceConfigOption func(*SourceConfig)
+
+// WithConcurrency sets the concurrency for the source.
+func WithConcurrency(concurrency int) SourceConfigOption {
+	return func(c *SourceConfig) {
+		c.Concurrency = concurrency
+	}
+}
+
+// WithVerify sets the verify flag for the source.
+func WithVerify(verify bool) SourceConfigOption {
+	return func(c *SourceConfig) {
+		c.Verify = verify
+	}
+}
+
+// WithSimpleCache sets the cache for the source.
+func WithSimpleCache() SourceConfigOption {
+	return func(c *SourceConfig) {
+		c.Cache = cache.Simple()
+	}
+}
+
+// SourceConfig provides a way to configure a Source.
+type SourceConfig struct {
```

Comment:
**Nit:** Thoughts on renaming this `Config` since it's already in the `source` package?